src/main/java/org/junitpioneer/jupiter/StdIoExtension.java

```diff
@@ -20,21 +20,38 @@
 import java.io.StringReader;
 import java.io.StringWriter;
 import java.io.Writer;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.List;
 
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeTestExecutionCallback;
+import org.junit.jupiter.api.extension.ExtensionConfigurationException;
 import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
 import org.junit.jupiter.api.extension.ParameterContext;
-import org.junit.jupiter.api.extension.ParameterResolutionException;
 import org.junit.jupiter.api.extension.ParameterResolver;
 
 /**
  * <p>For details and examples, see
  * <a href="https://junit-pioneer.org/docs/standard-input-output/" target="_top">the documentation on <code>Standard input/output</code></a>
  * </p>
  */
-public class StdIOExtension implements ParameterResolver {
+public class StdIoExtension implements ParameterResolver, BeforeTestExecutionCallback, AfterEachCallback {
 
 	private static final String SEPARATOR = System.getProperty("line.separator");
 
+	private static final Namespace NAMESPACE = Namespace.create(DefaultLocaleExtension.class);
+
+	private static final String IN_KEY = "StdIo_In";
+	private static final String OUT_KEY = "StdIo_Out";
+
+	@Override
+	public void afterEach(ExtensionContext context) {
+		System.setIn(context.getStore(NAMESPACE).get(IN_KEY, InputStream.class)); //NOSONAR resetting input
+		System.setOut(context.getStore(NAMESPACE).get(OUT_KEY, PrintStream.class));
```

Comment:
Could this break if only one of `StdIn` or `StdOut` is used?
`System.in` or `System.out` may end up being `null` if one of these `.get(...)` calls returns null.

Reply:
Yeah, should probably check if it even needs to be reset first.

Reply:
I added more tests to cover this case and changed the behaviour.